skops/io/_numpy.py

```diff
@@ -47,6 +48,48 @@ def ndarray_get_instance(state, src):
     return val
 
 
+def random_state_get_state(obj, dst):
+    content = get_state(obj.get_state(legacy=False), dst)
+    res = {
+        "__class__": obj.__class__.__name__,
+        "__module__": get_module(type(obj)),
+        "type": "numpy",
+        "content": content,
+    }
+    return res
```

Comment:
@rgommers would this be enough to save the state of a random state? Same with a generator bellow.

This is in the context of saving sklearn-compatible models while not using pickle.

Reply:
I _think_ that `bit_generator.state`, which seems to be used here, should be enough (hard to be certain without trying of course). And I'm assuming that, like pickle, the same numpy version is used to load the data in again.

Reply:
Oh that's interesting. For sklearn we need to use the same sklearn version to load things since things might move around. Should we be imposing the same restriction/warning for numpy/scipy? I thought `npy` format was more stable and we could safely kinda ignore the numpy version @rgommers .

sklearn gives a warning if the user tries to load using a different version, but I haven't seen numpy do that.

Reply:
> hard to be certain without trying of course

Would you say the [test](https://github.com/BenjaminBossan/skops/blob/aae23aa693f496d30c6615b411d86c505f9ff297/skops/io/tests/test_persist.py#L292-L303) is sufficient or could it be more comprehensive to cover some edge cases?

Theoretically, we could also test different numpy versions on CI but I think that would blow up our test matrix size, so it's unlikely.

Testing _different_ numpy versions for saving and loading seems quite hard to achieve. It would be good to know if numpy gives any guarantees there, but I didn't see anything in the docs.

Reply:
> I thought `npy` format was more stable and we could safely kinda ignore the numpy version @rgommers .

The `npy` format is indeed stable and versioned, and does not depend on the `numpy` version. However, (a) I didn't/don't see that used here, and (b) you seem to rely on the `BitGenerator` state, which AFAIK doesn't have guarantees as strong as the `.npy` format. xref https://numpy.org/neps/nep-0019-rng-policy.html for that